polus-feature-extraction-plugin/src/utility.py

```diff
@@ -0,0 +1,1035 @@
+                if csvfile == 'csvone':
+                    df_feature = df_feature.append(df)#append the dataframe to save all in one csv file
+                else:
+                    df_feature = df#assign the dataframe to save as separate file
+                index+=1
+                print("Number of images processed-",index)
+        #If intensity image is passed as a parameter        
+        else:
+            os.chdir(self.segment_dir)#change directory to get all segmented images
+            seg_filenames1 = glob.glob("*.ome.tif")#read only the .ome.tif files in the directory
+        
+            os.chdir(self.intensity_dir)#change directory to get all intensity images
+            intensity_filenames = glob.glob("*.ome.tif")#read only the .ome.tif files in the directory
+            
+            for seg_file_names1 in seg_filenames1:#run analysis for each segmented image in the list
+                intensity =difflib.get_close_matches(seg_file_names1, intensity_filenames,n=1, cutoff=0.1)#match the filename in segmented image to the  list of intensity image filenames to match the files
+                intensity = str(intensity[0])#get the filename of intensity image that has closest match
+                intensity_file = self.intensity_dir +"/"+ intensity#set the entire path for the bioreader to read the image
+                
+                #Read the intensity image using bioreader from bfio
+                br_int = BioReader(intensity_file)
+                intensity_bfio = br_int.read_image()
+                intensity_image= np.squeeze(intensity_bfio)
+                
+                #Read the segmented image using bioreader from bfio
+                seg_file= self.segment_dir + "/" + seg_file_names1#set the entire path for the bioreader to read the image
+                br_seg= BioReader(seg_file)
+                segment_bfio = br_seg.read_image()
+                seg_file = np.squeeze(segment_bfio)
+                label_image= self.labeling(seg_file)#labels the objects
+                
+                #Call the feature_extraction function in Analysis class
+                analysis = Analysis(label_image,features,seg_file_names1, csvfile,outDir,boxSize,angleDegree, pixelDistance,intensity_image)
+                df = analysis.feature_extraction()
+                
+                #Check whether csvfile is csvone to save the features extracted from all the images in same csv file
+                if csvfile == 'csvone':
+                    df_feature = df_feature.append(df)
+                else:
+                    df_feature = df
+                index+=1
+                print("Number of images processed-",index)
+
+
+        jutil.kill_vm()#kill the vm
+        return(df_feature,seg_filenames1)#return dataframe and the segment imag filename
+        
+#create class to extract features from the images      
+class Analysis(ConvertImage):
```

Comment:
Add class and method commenting according to the Google Python commenting specification.